File: samples/UpdateProperties/UpdateProperties/ViewController.m
//
//  ViewController.m
//  UpdateProperties
//
//  Created by Takeshi Shoji on 2016/08/17.
//  Copyright © 2016年 Novars Inc. All rights reserved.
//

#import "ViewController.h"
@import MaBeeeSDK;

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITextView *textView;

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

- (void)viewDidAppear:(BOOL)animated {
  [super viewDidAppear:animated];
  [MaBeeeApp.instance addObserver:self selector:@selector(receiveNotification:)];
}

- (void)viewWillDisappear:(BOOL)animated {
  [super viewWillDisappear:animated];
  [MaBeeeApp.instance removeObserver:self];
}

- (IBAction)scanButtonPressed:(UIButton *)sender {
  MaBeeeScanViewController *vc = MaBeeeScanViewController.new;
  [vc show:self];
}

- (IBAction)updateButtonPressed:(UIButton *)sender {
  for (MaBeeeDevice *device in MaBeeeApp.instance.devices) {
    [device updateRssi];
    [device updateBatteryVoltage];
  }
}

- (void)receiveNotification:(NSNotification *)notification {

  if ([MaBeeeDeviceRssiDidUpdateNotification isEqualToString:notification.name]) {
    NSUInteger identifier = [notification.userInfo[@"MaBeeeDeviceIdentifier"] unsignedIntegerValue];
    MaBeeeDevice *device = [MaBeeeApp.instance deviceWithIdentifier:identifier];
    NSString *line = [NSString stringWithFormat:@"%@ RSSI: %d", device.name, device.rssi];
    [self appendLine:line];
    return;
  }

  if ([MaBeeeDeviceBatteryVoltageDidUpdateNotification isEqualToString:notification.name]) {
    NSUInteger identifier = [notification.userInfo[@"MaBeeeDeviceIdentifier"] unsignedIntegerValue];
    MaBeeeDevice *device = [MaBeeeApp.instance deviceWithIdentifier:identifier];
    NSString *line = [NSString stringWithFormat:@"%@ Volgate: %f", device.name, device.batteryVoltage];
    [self appendLine:line];
    return;
  }
}

- (void)appendLine:(NSString *)line {
  self.textView.text = [self.textView.text stringByAppendingFormat:@"%@\n", line];
}

@end

File: samples/UpdatePropertiesSwift/UpdatePropertiesSwift/ViewController.swift
//
//  ViewController.swift
//  UpdatePropertiesSwift
//
//  Created by Takeshi Shoji on 2016/08/18.
//  Copyright © 2016年 Novars Inc. All rights reserved.
//

import UIKit
import MaBeeeSDK

class ViewController: UIViewController {

  @IBOutlet weak var textView: UITextView!

  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view, typically from a nib.
  }

  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }

  override func viewDidAppear(_ animated: Bool) {
    super.viewDidAppear(animated)
    MaBeeeApp.instance().addObserver(self, selector: #selector(ViewController.receiveNotification(_:)))
  }

  override func viewWillDisappear(_ animated: Bool) {
    super.viewWillDisappear(animated)
    MaBeeeApp.instance().removeObserver(self)
  }

  func receiveNotification(_ notification: Notification) {
    switch notification.name {
    case NSNotification.Name.MaBeeeDeviceRssiDidUpdate:
      if let identifier = notification.userInfo?["MaBeeeDeviceIdentifier"] as? UInt,
        let device = MaBeeeApp.instance().device(withIdentifier: identifier) {
          appendLine(device.name + " RSSI : " + String(device.rssi))
        }
    case NSNotification.Name.MaBeeeDeviceBatteryVoltageDidUpdate:
      if let identifier = notification.userInfo?["MaBeeeDeviceIdentifier"] as? UInt,
        let device = MaBeeeApp.instance().device(withIdentifier: identifier) {
        appendLine(device.name + " Volgate : " + String(device.batteryVoltage))
      }
    default:
      print(notification.name)
    }
  }

  func appendLine(_ line: String) {
    textView.text = textView.text + line + "\n"
  }

  @IBAction func scanButtnPressed(_ sender: UIButton) {
    let vc = MaBeeeScanViewController()
    vc.show(self)
  }

  @IBAction func updateButtonPressed(_ sender: UIButton) {
    for device in MaBeeeApp.instance().devices() {
      device.updateRssi()
      device.updateBatteryVoltage()
    }
  }
}

